src/commands/deployImage/imageSource/EnvironmentVariablesListStep.ts

```diff
@@ -4,32 +4,93 @@
     public async prompt(context: ImageSourceBaseContext): Promise<void> {
-        const input = await context.ui.showQuickPick([{ label: localize('set', 'Set with environment variable file') }, { label: skipForNowLabel }],
-            { placeHolder: localize('setEnvVar', 'Set environment variables in container instance') });
-
-        if (input.label !== skipForNowLabel) {
-            const envData = await this.selectEnvironmentSettings(context);
+        const envData: DotenvParseOutput | undefined = await this.selectEnvironmentSettings(context);
+        if (!envData) {
+            context.environmentVariables = [];
+            this.outputLogs(context, SetEnvironmentVariableOption.SkipForNow);
+        } else {
             context.environmentVariables = Object.keys(envData).map(name => { return { name, value: envData[name] } });
+            this.outputLogs(context, SetEnvironmentVariableOption.ProvideFile);
+        }
+    }
+
+    public async configureBeforePrompt(context: ImageSourceBaseContext): Promise<void> {
+        if (context.environmentVariables?.length === 0) {
+            this.outputLogs(context, SetEnvironmentVariableOption.NoDotEnv);
         }
     }
 
     public shouldPrompt(context: ImageSourceBaseContext): boolean {
         return context.imageSource !== ImageSource.QuickStartImage && context.environmentVariables === undefined;
     }
 
-    private async selectEnvironmentSettings(context: ImageSourceBaseContext) {
-        const envFileFsPath: string = await selectWorkspaceFile(context, 'Select a .env file', { filters: { 'env file': ['env', 'env.*'] } }, '**/*.{env,env.*}');
+    private async selectEnvironmentSettings(context: ImageSourceBaseContext): Promise<DotenvParseOutput | undefined> {
+        const placeHolder: string = localize('setEnvVar', 'Select a {0} file to set the environment variables for the container instance', '.env');
+        const envFileFsPath: string | undefined = await selectWorkspaceFile(context, placeHolder,
+            { filters: { 'env file': ['env', 'env.*'] }, allowSkip: true }, allEnvFilesGlobPattern);
+
+        if (!envFileFsPath) {
+            return undefined;
+        }
+
         const data = await AzExtFsExtra.readFile(envFileFsPath);
         return parse(data);
     }
+
+    public static async workspaceHasEnvFile(): Promise<boolean> {
+        const envFileUris: Uri[] = await workspace.findFiles(allEnvFilesGlobPattern);
+        return !!envFileUris.length;
+    }
+
+    // Todo: It might be nice to add a direct command to update just the environment variables rather than having to suggest to re-run the entire command again
+    private outputLogs(_context: ImageSourceBaseContext, setEnvironmentVariableOption: SetEnvironmentVariableOption): void {
+        if (setEnvironmentVariableOption !== SetEnvironmentVariableOption.ProvideFile) {
+            // context.activityChildren?.push(
```

Comment:
Why did you comment out of the activity log stuff? Not hooked up yet?

Reply:
Yup, going to hook up all the activity children logic in one PR later down the line because it's a bit more involved than just these item pushes... figured it's easier to review together in one wombo combo